### PR TITLE
fix(webapp): match org invite emails case-insensitively

### PR DESCRIPTION
**Pull request by isshaddad**
Org member invites couldn't be accepted when the invite email's casing didn't match the invitee's account email (for example, `User@example.com` vs `user@example.com`). The accept route compared emails strictly, and the pending-invite lookups used exact matching, so the invite never appeared for the invitee in the first place.

### CODE REVIEW

File: apps/webapp/app/models/member.server.ts [hunk 1]
```diff
@@ -110,35 +110,38 @@ export async function inviteMembers({
...
+          inviterId: userId,
+          role: "MEMBER",
+          rbacRoleId: rbacRoleId ?? null,
+        },
+        include: {
+          organization: true,
+          inviter: true,
+        },
+      })
+    )
```
[the remark this answers is not in the document]
Comment (isshaddad) [reply apps/webapp/app/models/member.server.ts:143]: This only affects a pre-existing mixed-case invite that's re-invited after this change. New invites are all lowercased, so they upsert cleanly.

The impact is limited to a transient duplicate row and potentially an extra email. `acceptInvite` already calls `deleteMany` (`member.server.ts:220`), which cleans up any case-variant records when the invite is accepted.

Given the narrow edge case and low impact, I don't think it's worth adding an extra read on every invite. I'd leave it as-is.

File: apps/webapp/test/inviteEmailCase.test.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      return prismaHolder.client;
+    },
+    get $replica() {
+      if (!prismaHolder.client) {
+        throw new Error("test prisma not set");
+      }
+      return prismaHolder.client;
+    },
+  };
+});
```
[the remark this answers is not in the document]
Comment (isshaddad) [reply apps/webapp/test/inviteEmailCase.test.ts:34]: Not changing this one — the file follows the established convention for this area.

`apps/webapp/test/member.server.test.ts` mocks the same two modules (`~/services/rbac.server` at L19, `~/db.server` at L39), and this file was modelled on it directly. On location, `apps/webapp/test/` holds 450 test files versus 152 alongside source, so this is where webapp tests live in practice.

Worth noting the `~/db.server` mock isn't mocking the database. It's the wiring that injects the real testcontainer Prisma client into the module under test — the tests run against actual Postgres via `postgresTest`, which is what the no-mocks rule is asking for. The RBAC mock is a genuine stub, and there I'm matching the sibling file rather than diverging in a single new test.

Happy to move the whole invite test surface if we want to change the convention, but that seems like its own change rather than something to do in this PR.

File: apps/webapp/app/models/member.server.ts [hunk 3]
```diff
@@ -616,7 +616,7 @@ export async function declineInvite({
     //2. check for other invites
     const remainingInvites = await tx.orgMemberInvite.findMany({
       where: {
-        email: user.email,
+        email: { equals: user.email, mode: "insensitive" },
```
[the remark this answers is not in the document]
Comment (isshaddad) [reply apps/webapp/app/models/member.server.ts:644]: Correct about the rule — `apps/webapp/CLAUDE.md:120` does require the traced `$transaction` helper from `~/db.server`. Leaving it out of this PR though.

That `prisma.$transaction` call is pre-existing on `main`, not introduced here: this branch only changes the `email` filters inside the transaction body, no transaction lines. There are also around a dozen direct `prisma.$transaction` calls across `apps/webapp/app`, so `declineInvite` isn't an outlier — converting just this one would be unrelated scope in a PR that's been waiting on review since June.

Worth doing as its own pass across the callers.

File: apps/webapp/app/models/member.server.ts [hunk 4]
```diff
@@ -135,11 +135,25 @@ export async function inviteMembers({
...
+  );
+
+  const pendingInvites = await prisma.orgMemberInvite.findMany({
+    where: {
+      organizationId: org.id,
+      email: { in: boundedIn([...uniqueEmails]), mode: "insensitive" },
+    },
+    select: { email: true },
+  });
+  const pendingInviteEmails = new Set(pendingInvites.map((invite) => invite.email.toLowerCase()));
```
[the remark this answers is not in the document]
Comment (isshaddad) [reply apps/webapp/app/models/member.server.ts:156]: Good catch — fixed in 34f1f712e by writing the folded address rather than the one the caller passed.

The read-then-write window you describe was real: with rows able to differ by case, `@@unique([organizationId, email])` could never fire for a variant, so there was nothing behind the in-memory check. Storing folded makes every row for an address identical, which lets that constraint do the job it was already there for — the loser of a concurrent insert now gets P2002 and is reported as `alreadyInvited`.

It also stops `inviteMembers` depending on its callers to normalise first, so the dashboard and PAT schemas are now defence in depth instead of load-bearing.

Covered by a test that calls `inviteMembers` directly with mixed case, asserts the stored row is folded, and asserts a differently cased variant doesn't become a second row.

Not addressed here: existing mixed-case rows created before this change. Those still match correctly on accept and decline, since the lookups fold, but the constraint can't dedupe them against a new folded row. A backfill would need its own migration.

**Review by 1stvamp — APPROVED** [review]
